src/main/java/org/jabref/Globals.java

```diff
@@ -90,10 +91,13 @@ public static void startBackgroundTasks() {
     }
 
     private static void stopTelemetryClient() {
-        if (Globals.prefs.shouldCollectTelemetry()) {
-            getTelemetryClient().ifPresent(client -> client.trackSessionState(SessionState.End));
-            getTelemetryClient().ifPresent(client -> client.flush());
-        }
+        getTelemetryClient().ifPresent(client -> {
+            client.trackSessionState(SessionState.End);
+            client.flush();
+
+            // Workaround for bug https://github.com/Microsoft/ApplicationInsights-Java/issues/662
```

Comment:
Maybe add a FIXME here, so we can revert this when the bug fix is merged

Reply:
done